docs/book/src/plugins/extending-cli.md

```diff
@@ -0,0 +1,205 @@
+# Extending the CLI and Scaffolds
+
+## Overview
+
+You can extend Kubebuilder to allow your project to have the same CLI features and provide the plugins scaffolds.
+
+## CLI system
+
+Plugins are run using a [`CLI`][cli] object, which maps a plugin type to a subcommand and calls that plugin's methods.
+For example, writing a program that injects an `Init` plugin into a `CLI` then calling `CLI.Run()` will call the
+plugin's [SubcommandMetadata][plugin-sub-command], [UpdatesMetadata][plugin-update-meta] and `Run` methods with information a user has passed to the
+program in `kubebuilder init`. Following an example:
+
+```go
+package cli
+
+import (
+    log "github.com/sirupsen/logrus"
+    
+	"sigs.k8s.io/kubebuilder/v3/pkg/cli"
+	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	configbasev1 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/config-base/v1"
```

Comment:
@camilamacedo86 Sorry I didn't bring this up earlier... but I just discovered this while building a test plugin.  This pkg doesn't seem to exist.  Is this import path right?  I'm looking in your branch for PR #2106 : https://github.com/camilamacedo86/kubebuilder/tree/config-base/pkg/plugins

Reply:
We changed the path and I did not updated that here. It is : `kustomizecommonv1 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v1"`

Reply:
Done now 👍